File: core/src/main/java/ru/surfstudio/android/core/ui/base/placeholder/PlaceHolderView.java
package ru.surfstudio.android.core.ui.base.placeholder;

import ru.surfstudio.android.core.ui.base.screen.model.state.LoadState;

/**
 * Интрефейс для вью отрисовывающей состояние {@link LoadState}
 */
public interface PlaceHolderView {

    void render(LoadState loadState);
}

File: filestorage-sample/src/main/java/ru/surfstudio/android/filestorage/sample/ui/base/placeholder/PlaceHolderView.kt
package ru.surfstudio.android.filestorage.sample.ui.base.placeholder

import android.content.Context
import android.util.AttributeSet
import ru.surfstudio.android.core.mvp.model.state.LoadState
import ru.surfstudio.android.core.mvp.placeholder.PlaceHolderViewInterface
import ru.surfstudio.android.custom.view.placeholder.StandardPlaceHolderView

class PlaceHolderView(context: Context, attributeSet: AttributeSet
) : StandardPlaceHolderView(context, attributeSet), PlaceHolderViewInterface {

    override fun render(loadState: LoadState?) {
        when (loadState) {
            LoadState.NONE -> setNoneState()
            LoadState.MAIN_LOADING -> setMainLoadingState()
            LoadState.TRANSPARENT_LOADING -> setTransparentLoadingState()
            LoadState.EMPTY -> setEmptyState()
            LoadState.ERROR -> setErrorState()
            LoadState.NOT_FOUND -> setNotFoundState()
            LoadState.NO_INTERNET -> setNoInternetState()
        }
    }
}